app/controllers/cities_controller.rb

```diff
@@ -0,0 +1,35 @@
+class CitiesController < ApplicationController
+  def index
+    @cities = City.all
+  end
+
+  def new
+    if user_signed_in?
+      @city = City.new
+    else
+      flash[:notice] = "You need to be signed in to add a City."
+      redirect_to new_user_session_path
+    end
+  end
+
+  def create
+    @city = City.new(city_params.merge(user_id: current_user.id))
+    unless user_signed_in?
```

Comment:
we should only run the conditional below *if* the user is signed in. Execution will not stop on the `redirect`